python/lsst/consdb/hinfo_fix_missing.py

```diff
@@ -0,0 +1,264 @@
+
+        # Convert from RA and Dec
+        s_ra, s_dec = map(lambda x: float(exposure_rec[x]), ("s_ra", "s_dec"))
+        location = EarthLocation.of_site("LSST")
+        obstimes = Time(
+            [
+                exposure_rec["obs_start_mjd"],
+                exposure_rec["exp_midpt_mjd"],
+                exposure_rec["obs_end_mjd"],
+            ],
+            format="mjd",
+            scale="tai",
+        )
+        coord = SkyCoord(s_ra, s_dec, unit="deg")
+        altaz = coord.transform_to(AltAz(obstime=obstimes, location=location))
+
+        # Get altaz calculations
+        altitude_start, altitude, altitude_end = altaz.alt.deg
+        azimuth_start, azimuth, azimuth_end = altaz.az.deg
+        zenith_distance_start, zenith_distance, zenith_distance_end = altaz.zen.deg
+
+        # Use K&Y 1989 model to compute airmass from altitude
+        airmass = None
+        if altitude >= 0 and altitude <= 90:
+            a_ky89 = 0.50572
+            b_ky89 = 6.07995
+            c_ky89 = 1.6364
+            airmass = 1 / (sin(radians(altitude)) + a_ky89 * (altitude + b_ky89) ** -c_ky89)
+
+        # Load them into the update dictionary.
+        update = {}
+        calculations = {
+            "altitude_start": altitude_start,
+            "altitude": altitude,
+            "altitude_end": altitude_end,
+            "azimuth_start": azimuth_start,
+            "azimuth": azimuth,
+            "azimuth_end": azimuth_end,
+            "zenith_distance_start": zenith_distance_start,
+            "zenith_distance": zenith_distance,
+            "zenith_distance_end": zenith_distance_end,
+            "airmass": airmass,
+        }
+        for key, value in calculations.items():
+            if exposure_rec[key] is None and value is not None:
+                update[key] = value
+                self.logger.debug(f"Inferring column: {key}")
+        return update
+
+    def fix_band(self, exposure_rec: dict[str, Any]) -> dict[str, Any]:
+        """Tries to identify band, if not provided.
+
+        This function relies on the physical_filter column to
+        indicate the band. It uses two formats:
+         * u_01 => u band, seen in LSSTComCam
+         * SDSSr_65mm~empty => r band, seen in LATISS
+
+        It modifies the `exposure_rec` dictionary and returns
+        nothing.
+
+        Parameters
+        ----------
+        exposure_rec : `dict[str, Any]`
+            The exposure record dictionary, (almost) ready to
+            be copied into consolidated database exposure
+            table.
+
+        Returns
+        -------
+        dict[str, Any]
+            Fixed columns in the original `exposure_rec` dictionary.
+        """
+        if exposure_rec["band"] is not None:
+            # Band already set
+            return dict()
+
+        if exposure_rec["physical_filter"] is None:
+            # Can't infer band from physical_filter
+            return dict()
+
+        for band in "ugrizy":
```

Comment:
Seems better to match `r"SDSS([ugrizy])"` or `r"([ugrizy])_\d+"` and then extract the group.  Then no loop is needed.